lib/chef/provider/package/apt.rb

```diff
@@ -71,15 +71,11 @@ def candidate_version
         end
 
         def package_locked(name, version)
-          islocked = false
           locked = shell_out_compact_timeout!("apt-mark", "showhold")
-          locked.stdout.each_line do |line|
-            line_package = line.strip
-            if line_package == name
-              islocked = true
-            end
+          locked_packages = locked.stdout.each_line.map do |line|
+            line.strip
           end
-          islocked
+          name.all? { |n| locked_packages.include? n }
```

Comment:
Small nitpick: I'd love to see this be `include?(n)` so we remain consistent with method params being wrapped in parentheses. Not a blocker for this, just something to think about for your next change.